Src/MakingSense.SafeBrowsing/GoogleSafeBrowsing/GoogleSafeBrowsingUpdater.cs

```diff
@@ -0,0 +1,141 @@
+            if (!_database.AllowRequest)
+            {
+                return;
+            }
+
+            var request = new FetchThreatListUpdatesRequest
+            {
+                Client = new ClientInfo
+                {
+                    ClientId = _configuration.ClientId,
+                    ClientVersion = _configuration.ClientVersion
+                },
+                ListUpdateRequests = _database.SuspiciousLists.Select(list => new ListUpdateRequest
+                {
+                    PlatformType = ANY_PLATFORM,
+                    ThreatType = list.Key,
+                    ThreatEntryType = URL,
+                    State = list.Value.State,
+                    Constraints = new Constraints
+                    {
+                        Region = _configuration.Region,
+                        SupportedCompressions = new List<string>
+                    {
+                        RAW
+                    }
+                    }
+                }).ToList()
+            };
+
+            var response = await _safebrowsingService.ThreatListUpdates.Fetch(request).ExecuteAsync();
+
+            _database.MinimumWaitDuration = TimeSpan.FromSeconds(double.Parse((response.MinimumWaitDuration as string).TrimEnd('s')));
+            _database.Updated = DateTimeOffset.Now;
+
+            foreach (var listUpdate in response.ListUpdateResponses)
+            {
+                if(listUpdate.ResponseType == FULL_UPDATE)
+                {
+                    _database.SuspiciousLists[listUpdate.ThreatType] = new SafeBrowsingList
+                    {
+                        State = response.ListUpdateResponses.First().NewClientState
+                    };
+
+                    foreach (var add in listUpdate.Additions)
+                    {
+                        var rawHashes = add.RawHashes;
+                        var hashValues = Convert.FromBase64String(rawHashes.RawHashesValue);
+                        _database.SuspiciousLists[listUpdate.ThreatType].Hashes.AddRange(SplitByteList(hashValues, rawHashes.PrefixSize.Value));
+                    }
+
+                    //_database.UnsafeLists[listUpdate.ThreatType].Hashes.Sort();
+                }
+                else
+                {
+                    //TODO: implement partial updates
+                }
+            }
+        }
+
+        /// <summary>
+        /// Split an array of bytes in smaller arrays based on input size.
+        /// </summary>
+        /// <param name="bytes">Original long array of bytes</param>
+        /// <param name="size">New small arrays lenth</param>
+        /// <returns>List of splitted arrays of bytes</returns>
+        private List<byte[]> SplitByteList(byte[] bytes, int size)
+        {
+            var list = new List<byte[]>();
+            var aux = new List<byte>();
+
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                aux.Add(bytes[i]);
+
+                if (aux.Count == size)
```

Comment:
Should there be a more efficient way to split an array of bytes, maybe indicating _from_ and _to_ values. Maybe it is not the focus right now, in that case, please simply add a _// TODO_ comment indicating that.